examples/vumps/vumps_ising.jl

```diff
@@ -102,3 +102,72 @@ Sz2_infinite = expect(ψ.AL[2] * ψ.C[2], "Sz")
 
 @show Sz1_finite, Sz2_finite
 @show Sz1_infinite, Sz2_infinite
+
+##############################################################################
+#Extract a finite MPS from an InfiniteCanonicalMPS by slicing
+
+function finite_mps(ψ::InfiniteCanonicalMPS, range::AbstractRange)
+  @assert isone(step(range))
+  ψ_finite = ψ.AL[range]
+  ψ_finite[last(range)] *= ψ.C[last(range)]
+  l0 = linkind(ψ.AL, first(range) - 1 => first(range))
+  lN = linkind(ψ.AR, last(range) => last(range) + 1)
+  ψ_finite = MPS([δ(l0, dag(l0)'); [ψ_finiteᵢ for ψ_finiteᵢ in ψ_finite]; δ(dag(lN), lN')])
+  set_ortho_lims!(ψ_finite, (last(range) + 1):(last(range) + 1))
+  return ψ_finite
+end
+
+
+##########################################################
+#Correlation matrix for infinite MPS
+
+function correlation_matrix(ψ::InfiniteCanonicalMPS, op1, op2, dim)
+  C = zeros(ComplexF64, dim, dim)
+  for i in 1:dim
+      for j in (i+1):dim
+          h = op(op1, siteinds(ψ)[i]) * op(op2, siteinds(ψ)[j])
+          ϕ = ψ.AL[i]
+          for k in (i+1):(j)
+              ϕ *= ψ.AL[k]
+          end
+          ϕ *= ψ.C[j]
+          C[i,j] = (noprime(ϕ * h) * dag(ϕ))[]
+      end
+  end
+  return C + C'
+end
+
+##########################################################
+#Correlation matrix for finite MPS using explicit contraction of operator
+
+function correlation_matrix_gates(ψ::MPS, op1, op2, start, stop)
+  C = zeros(ComplexF64, stop-start+1, stop-start+1)
+  for i in start:stop
+      orthogonalize!(ψ, i)
+      for j in (i+1):stop
+          h = op(op1, siteinds(ψ)[i]) * op(op2, siteinds(ψ)[j])
+          ϕ = ψ[i]
+          for k in (i+1):(j)
+              ϕ *= ψ[k]
+          end
+          C[i-start+1,j-start+1] = (noprime(ϕ * h) * dag(ϕ))[]
+      end
+  end
+  return C + C'
+end
```

Comment:
I think we can just remove this and show how to use `ITensors.correlation_matrix`.

As a test we could just compare the result to a single correlator computed with an MPO.